transformer.py

```diff
@@ -68,15 +68,15 @@ def forward(self, x):
         # mask = mask.round().to(dtype=torch.int64)
         # masked_indices = torch.zeros_like(z_indices)
         masked_indices = self.mask_token_id * torch.ones_like(z_indices, device=z_indices.device)
-        a_indices = mask * z_indices + (~mask) * masked_indices
+        a_indices = ~mask * z_indices + (mask) * masked_indices
 
         a_indices = torch.cat((sos_tokens, a_indices), dim=1)
 
         target = torch.cat((sos_tokens, z_indices), dim=1)
 
         logits = self.transformer(a_indices)
 
-        return logits, target
+        return logits[~mask], target[~mask]

```

Comment:
This should be like this?
```suggestion
        return logits[:, ~mask], target[:, ~mask]

```